src/asset-loader.ts

```diff
@@ -48,7 +48,8 @@ class AssetLoader {
                     {
                         elementFilter: this.loadAllData ? (() => true) : null,
                         // decompress data on load
-                        decompress: true
+                        decompress: true,
+                        reorder: false, // Do not perform MortonOrder calculation and reordering when working with labels, since we are not persiting the old indices so we could also reorder the labels that come after.
```

Comment:
This should also be improved - the ordering of splats and their corresponding labels should persist even after MortonOrder is applied.